From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 23 Nov 2022 09:08:22 +0200
Subject: [PATCH 49/67] Follow ambience for prefers-color-scheme. JB#58394

Treat ui.systemUsesDarkTheme=2 as the EmbedLite follows-ambience sentinel
and fall through to the Qt native value. Keep explicit 0/1 preference values
authoritative.
---
 widget/nsXPLookAndFeel.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/widget/nsXPLookAndFeel.cpp b/widget/nsXPLookAndFeel.cpp
index 34b327e08b2d..9ef7a54e577a 100644
--- a/widget/nsXPLookAndFeel.cpp
+++ b/widget/nsXPLookAndFeel.cpp
@@ -1024,7 +1024,9 @@ Maybe<nscolor> nsXPLookAndFeel::GetUncachedColor(ColorID aID,
 
 nsresult nsXPLookAndFeel::GetIntValue(IntID aID, int32_t& aResult) {
   if (NS_SUCCEEDED(Preferences::GetInt(sIntPrefs[size_t(aID)], &aResult))) {
-    return NS_OK;
+    if (aID != IntID::SystemUsesDarkTheme || aResult != 2) {
+      return NS_OK;
+    }
   }
 
   if (NS_FAILED(NativeGetInt(aID, aResult))) {
